### PR TITLE
Adding Service Table

### PR DESCRIPTION
**Pull request by jumaffre**
Resolves https://github.com/microsoft/CCF/issues/294. 

This PR introduces a new `SERVICE` table that keeps track of the state of the service (as described in the paper). It maps from the version at which the service is valid from (starts at 0, updated for every recovery) to the certificate of the network and its status. 

The network should transition from `OPENING` to `OPEN` and then `CLOSED`. Note that until https://github.com/microsoft/CCF/issues/293 is complete, the service opens in `OPEN` straight away.

I've also added some checks in our python end-to-end infra for members to check the status of the network whenever is starts/restarts.

### CODE REVIEW

**Comment by cimetrics** [comment]
images

**Comment by cimetrics** [comment]
images

File: tests/infra/ccf.py [hunk 1]
```diff
@@ -25,6 +26,12 @@ class NodeNetworkState(Enum):
     joined = 2
 
 
+class ServiceStatus(Enum):
+    OPENING = 1
+    OPEN = 2
+    CLOSED = 2
```
Comment (eddyashton) [inline tests/infra/ccf.py]: 2 is used twice

**Comment by cimetrics** [comment]
images

**Comment by cimetrics** [comment]
images

**Comment by cimetrics** [comment]
images

**Pull request merged**